### ISSUE TITLE
Use category filtering AND load more

### ISSUE THREAD

**Issue by brycekirk** [putyourlightson/craft-sprig#145]
### Describe the bug
I have some checkboxes that filter entries by category (industry), and a load more button that is supposed to append additional entries to the query results. The issue is that the load more button disappears whenever a filter is applied, even if there are more entries.

Here's my simplified sprig component:
``` twig
{% set category = craft.categories.id(categoryId).one() %}
{% set industry = industry ?? '' %}
{% set offset = offset ?? 0 %}

{# This is a category page, so before any industry sorting happens, only templates related to the category should show #}
{% set relatedTo = ['and', categoryId] %}
{% set relatedTo = industry ? relatedTo|merge(industry) : categoryId %}

<div id="filters">
  <p>Industry:</p>
  <ul>
    {% for indCat in craft.categories.group('templateCategory').all() %}
      <li>
        <input type="checkbox" id="{{ indCat.slug }}" value="{{ indCat.id }}" name="industry[]" {{ indCat.id in industry ? 'checked' }} sprig>
        <label for="{{ indCat.slug }}">{{ indCat.title }}</label>
      </li>
    {% endfor %}
  </ul>
</div>

<div id="entries">
  {# limit = 6, sent with the sprig component include #}
  {% set entryQuery = craft.entries
    .section('proposalTemplates')
    .limit(limit) 
    .offset(offset)
    .relatedTo(relatedTo)
  %}
  {% set entries = entryQuery.all() %}

  {% if entries|length %}
    {% for entry in entries %}
      <div class="entry">
        {{ entry.title }}
      </div>
    {% endfor %}
  {% endif %}
</div>

{% if entryQuery.count() > (offset + entries|length) %}
  <button id="load-more-oob" sprig
    s-val:offset="{{ offset + limit }}"
    s-select="#entries .entry"
    s-target="#entries" 
    s-swap="beforeend"
    s-swap-oob="true">
    Load More
  </button>

{% else %}
  <button id="load-more-oob" s-swap-oob="true" style="display: none"></button>
{% endif %}
```

### Versions
- Plugin version: 1.7.0
- Craft version: Craft Pro 3.6.14

**Comment by bencroker** [comment]
> The issue is that the load more button disappears whenever a filter is applied, even if there are more entries.

What exactly do you mean by "whenever a filter is applied", in this case?

**Comment by brycekirk** [comment]
When a checkbox is clicked and sprig reloads entries. In this case, it's a category from the `templateCategory` group

**Comment by bencroker** [comment]
This is tough to test without recreating your exact setup, can you provide a URL that I can take a look at?

**Comment by brycekirk** [comment]
Yep, but it's a staging link so I'd prefer not to share it publicly

**Comment by bencroker** [comment]
Sure, email it to info@putyourlightson.com

**Comment by bencroker** [comment]
After replicating your setup locally I discovered the issue. If a checkbox is clicked then the entire component is re-rendered, but since the button still has a `s-swap-oob="true"` attribute set, it is being excluded. So the solution is to conditionally include that attribute.

```twig
{{ sprig.trigger == 'load-more-oob' ? 's-swap-oob="true"' }}
```

I've updated the recipe at https://putyourlightson.com/sprig/cookbook#load-more-complex to reflect this.
